scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -797,72 +885,102 @@ private ClickHandler clickHandlerForCheckButton() {
-                        if (result == null) {
-                            GWT.log("Unexpected empty result");
-                            displayErrorMessage("Unexpected empty result");
-                            return;
-                        }
-
-                        if (result.startsWith(SubmitEditServlet.ERROR)) {
-                            displayErrorMessage(result.substring(SubmitEditServlet.ERROR.length()));
+                        if (isResultNull(result) || ifResultIsAnError(result)) {
                             return;
                         }
 
                         try {
                             JSONValue json = JSONParser.parseStrict(result);
                             JSONObject obj = json.isObject();
-                            if (obj != null) {
-                                if (obj.containsKey("valid")) {
-                                    if (((JSONBoolean) obj.get("valid")).booleanValue()) {
-                                        if (obj.containsKey("updatedVariables")) {
-                                            updateVariables(obj.get("updatedVariables"));
-                                            redrawVariables();
-                                        }
-                                        GWT.log("Job validated");
-                                        displayInfoMessage("Job is valid");
-                                    } else if (obj.containsKey("errorMessage")) {
-                                        String errorMessage = obj.get("errorMessage").toString();
-                                        if (errorMessage.contains("JobValidationException")) {
-                                            errorMessage = errorMessage.substring(errorMessage.indexOf(":") + 2);
-                                        }
-                                        displayErrorMessage(errorMessage);
+                            handleVariables(obj);
+                        } catch (Throwable t) {
+                            handleParsingError(result);
+                        }
+                    }
+
+                    private void handleParsingError(String result) {
+                        // JSON parsing error workaround to force extract error message
+                        MatchResult errorMessageMatcher = RegExp.compile(ERROR_MESSAGE_REGEX).exec(result);
+                        if (errorMessageMatcher != null) {
+                            GWT.log(errorMessageMatcher.getGroup(1));
+                            displayErrorMessage(errorMessageMatcher.getGroup(1));
+                        } else {
+                            GWT.log(JSON_ERROR);
+                            displayErrorMessage(JSON_ERROR);
+                        }
+                    }
+
+                    private void handleVariables(JSONObject obj) {
+                        if (obj != null) {
+                            if (obj.containsKey("valid")) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)